src/test/java/org/elasticsearch/index/query/MatchAllQueryBuilderTest.java

```diff
@@ -22,17 +22,20 @@
 import org.apache.lucene.search.MatchAllDocsQuery;
 import org.apache.lucene.search.Query;
 
-import java.io.IOException;
-
-import static org.hamcrest.Matchers.instanceOf;
-import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.equalTo;
 
 public class MatchAllQueryBuilderTest extends BaseQueryTestCase<MatchAllQueryBuilder> {
 
     @Override
-    protected void assertLuceneQuery(MatchAllQueryBuilder queryBuilder, Query query, QueryParseContext context) throws IOException {
-        assertThat(query, instanceOf(MatchAllDocsQuery.class));
-        assertThat(query.getBoost(), is(queryBuilder.boost()));
+    protected Query createExpectedQuery(MatchAllQueryBuilder queryBuilder, QueryParseContext context) {
+        MatchAllDocsQuery matchAllDocsQuery = new MatchAllDocsQuery();
+        matchAllDocsQuery.setBoost(queryBuilder.boost());
+        return matchAllDocsQuery;
+    }
+
+    @Override
+    protected void assertLuceneQuery(MatchAllQueryBuilder queryBuilder, Query query, QueryParseContext context) {
+        assertThat(query.getBoost(), equalTo(queryBuilder.boost()));
```

Comment:
I wonder if we need to test the boost if it is already part of the expected lucene query we check in the base test. That change moves a lot of the test code up to the base test. However, would be good to not completely get rid of the abstract assertLuceneQuery for now since we still might have checks here for specific builders. This boost check however is covere, I think.